util/reggen/reg_block.py

```diff
@@ -419,3 +424,10 @@ def make_alert_regs(self, alerts: List[Alert]) -> None:
     def get_addr_width(self) -> int:
         '''Calculate the number of bits to address every byte of the block'''
         return (self.offset - 1).bit_length()
+
+    def set_reg_shadowed(self) -> None:
+        '''Return boolean indication whether reg block contains shadowed reigsters'''
+        for r in self.flat_regs:
+            if r.shadowed:
+                self.shadowed = True
+                return
```

Comment:
Could we just do the check each time, giving a function called something like `has_shadowed_reg()` and not storing `self.shadowed` at all? I doubt the performance impact is measurable, and it's one less bit of state to get out of sync.

Reply:
sounds good, i can do that.